src/parallel/MPI_TIMED/ProcessTimer.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * ProcessTimer.h
+ *
+ *  Created on: Apr 12, 2020
+ *      Author: jeremyharisch
+ */
+
+#ifndef MARDYN_PROCESSTIMER_H
+#define MARDYN_PROCESSTIMER_H
+
+#include <array>
+#include <fstream>
+#include <map>
+#include <vector>
+
+#include <mpi.h>
+
+class ProcessTimer {
+public:
+	static ProcessTimer& get() {
+		static ProcessTimer processTimer;
+		return processTimer;
+	}
+
+	/*! @brief Switches between the profiling Levels
+	 *  This Fnction should be called right before and after a corresponding MPI-Function-Call
+	 *  to exclude only this one
+	 *
+	 *  @param level ==1 Profiling is enabled(default); ==0 Profiling is disabled
+	 */
+	int switchProfiling(const int level = 1) {
+		_profiling_switch = level;
+		return _profiling_switch;
+	}
+
+	//! @brief Starts Timer for MPI-Measurement
+	//!
+	//! @param process Rank of process
+	void startTimer() {
+		if (_profiling_switch == 1) {
+			double measurement_time = MPI_Wtime();
+			_process_time -= measurement_time;
+			_processes_debug.push_back(-measurement_time);
+		}
+	}
+
+	//! @brief Stops Timer for MPI-Measurement
+	void stopTimer() {
+		if (_profiling_switch == 1) {
+			double measurement_time = MPI_Wtime();
+			_process_time += measurement_time;
+			_processes_debug[_processes_debug.size() - 1] -= measurement_time;
```

Comment:
```suggestion
			_processes_debug[_processes_debug.size() - 1] += measurement_time;
```
This is meant to store all individual time measurements, right?